clientv3/auth.go

```diff
@@ -146,3 +138,25 @@ func StrToPermissionType(s string) (PermissionType, error) {
 	}
 	return PermissionType(-1), fmt.Errorf("invalid permission type: %s", s)
 }
+
+type Authenticator interface {
+	// Authenticate does authenticate with given user name and password.
+	Authenticate(ctx context.Context, name string, password string) (*AuthenticateResponse, error)
+}
+
+type authenticator struct {
+	conn   *grpc.ClientConn // conn in-use
+	remote pb.AuthClient
+}
+
+func (auth *authenticator) Authenticate(ctx context.Context, name string, password string) (*AuthenticateResponse, error) {
+	resp, err := auth.remote.Authenticate(ctx, &pb.AuthenticateRequest{Name: name, Password: password})
+	return (*AuthenticateResponse)(resp), rpctypes.Error(err)
+}
+
+func NewAuthenticator(conn *grpc.ClientConn) Authenticator {
```

Comment:
Because they don't have opts. opts is constructed in `Client.Dial()` and has an information e.g. about TLS. Simply passing the opts constructed in the function to authenticator would be the simplest way.